server/api/dataset.py

```diff
@@ -242,3 +248,55 @@ def submitReviewImages(self, dataset, params):
 
         # TODO: return value?
         return {'status': 'success'}
+
+    @describeRoute(
+        Description('Register metadata with a dataset.')
+        .param('id', 'The ID of the dataset.', paramType='path')
+        .param('uploadFolderId', 'The ID of the folder that contains metadata '
```

Comment:
Change this into `metadataFileId`, and have it point to a single Girder file, which should greatly simplify the validation here.